cuda/solver/common_trs_kernels.cuh

```diff
@@ -366,9 +368,15 @@ __global__ void sptrsv_naive_caching_kernel(
     const IndexType* const rowptrs, const IndexType* const colidxs,
     const ValueType* const vals, const ValueType* const b, size_type b_stride,
     ValueType* const x, size_type x_stride, const size_type n,
-    const size_type nrhs, bool* nan_produced)
+    const size_type nrhs, bool* nan_produced, IndexType* atomic_counter)
 {
-    const auto full_gid = thread::get_thread_id_flat<IndexType>();
+    __shared__ IndexType block_base_idx;
+    if (threadIdx.x == 0) {
+        block_base_idx =
+            atomic_add(atomic_counter, IndexType{1}) * default_block_size;
```

Comment:
Doesn't `block_base_idx` need to be marked as volatile, or otherwise reads to it could be optimized away? Though this behavior does not seem to occur.

Reply:
what do you mean here? the block_base_idx is only shared in thread-block, and there is a `__syncthreads()` so I think it should be okay?